File: app/src/main/java/com/aavens/mindloft/ui/main/RoomActionsDialogFragment.java
package com.aavens.mindloft.ui.main;

import android.app.AlertDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.content.Context;
import android.content.DialogInterface;
import android.net.Uri;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.aavens.mindloft.R;
import com.aavens.mindloft.listeners.DialogFragmentActionsListener;


public class RoomActionsDialogFragment extends DialogFragment {

    private static final String INDEX_ARG = "INDEX_ARG";

    private DialogFragmentActionsListener listener;
    private Integer index;

    public static RoomActionsDialogFragment newInstance(Integer index, DialogFragmentActionsListener listener) {
        RoomActionsDialogFragment fragment = new RoomActionsDialogFragment();
        fragment.listener = listener;
        Bundle args = new Bundle();
        args.putInt(INDEX_ARG, index);
        fragment.setArguments(args);
        return fragment;
    }

    public RoomActionsDialogFragment() {
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        index = getArguments().getInt(INDEX_ARG);
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setTitle(R.string.choose_dialog_title);
        builder.setItems(R.array.room_actions_array, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                listener.onActionInteraction(index);
            }
        });
        return builder.create();
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof DialogFragmentActionsListener) {
            listener = (DialogFragmentActionsListener) context;
        } else {
            throw new RuntimeException(context.toString()
                    + " must implement OnFragmentInteractionListener");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        listener = null;
    }
}

File: app/src/main/java/com/aavens/mindloft/ui/main/RoomActionsDialogFragment.kt
package com.aavens.mindloft.ui.main

import android.app.Dialog
import android.app.DialogFragment
import android.content.Context
import android.content.DialogInterface
import android.os.Bundle
import android.support.v7.app.AlertDialog
import com.aavens.mindloft.R
import com.aavens.mindloft.listeners.DialogFragmentActionsListener

class RoomActionsDialogFragment(index: Int, listener: DialogFragmentActionsListener?): DialogFragment() {

    constructor() : this(0, null) {
    }

    init {
        val args = Bundle()
        args.putInt(Args.index, index)
        arguments = args
    }

    private object Args {
        val index: String = "INDEX_ARG";
    }
    private var index: Int = index
    private var listener: DialogFragmentActionsListener? = listener

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        index = arguments.getInt(Args.index);
        val builder = AlertDialog.Builder(activity);
        builder.setTitle(R.string.choose_dialog_title);
        builder.setItems(
                R.array.room_actions_array,
                { dialogInterface: DialogInterface, i: Int -> listener?.onActionInteraction(index) })
        return builder.create()
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is DialogFragmentActionsListener) {
            listener = context
        } else {
            throw RuntimeException(context.toString() + " must implement OnFragmentInteractionListener")
        }
    }

    override fun onDetach() {
        super.onDetach();
        listener = null;
    }
}